src/modules/Modal/Modal.js

```diff
@@ -175,10 +175,8 @@ class Modal extends Component {
   handleClose = (e) => {
     debug('close()')
 
-    const { onClose } = this.props
-    if (onClose) onClose(e, this.props)
-
-    this.trySetState({ open: false }, { scrolling: false })
+    _.invoke(this.props, 'onClose', e, this.props)
+    this.trySetState({ open: false })
```

Comment:
My initial suggestion [there](https://github.com/Semantic-Org/Semantic-UI-React/issues/2150#issuecomment-333460199) was falsy. It causes state after `onClose` was done.

The same effect can be archived when `scrolling` will be reseted in `handleMount`.